### PR TITLE
Modify context menu options for executable and installer files

### LINKED ISSUE #606: Add a run as admin option for .exe files

### PR DESCRIPTION
**Pull request by Jaiganeshkumaran**
Implements and closes #606.

### CODE REVIEW

File: Files/Strings/en-US/Resources.resw [hunk 1]
```diff
@@ -753,4 +753,7 @@
   <data name="ByteSymbol" xml:space="preserve">
     <value>B</value>
   </data>
+  <data name="BaseLayoutItemContextRunAsAdminItem" xml:space="preserve">
```
Comment (yaichenbaum) [inline Files/Strings/en-US/Resources.resw:756]:
```suggestion
  <data name="BaseLayoutItemContextRunAsAdminItem.Text" xml:space="preserve">
```

File: Files/Views/LayoutModes/GenericFileBrowser.xaml [hunk 2]
```diff
@@ -234,6 +234,16 @@
                     <FontIcon Glyph="&#xEE3F;" />
                 </MenuFlyoutItem.Icon>
             </MenuFlyoutItem>
+            <MenuFlyoutItem
+                x:Name="RunAsAdminItem"
+                x:Uid="BaseLayoutItemContextRunAsAdminItem"
```
Comment (yaichenbaum) [inline Files/Views/LayoutModes/GenericFileBrowser.xaml:239]:
```suggestion
                x:Name="RunAsAdmin"
                x:Uid="BaseLayoutItemContextRunAsAdmin"
```

File: Files/Views/LayoutModes/GridViewBrowser.xaml [hunk 3]
```diff
@@ -230,6 +230,16 @@
                     <FontIcon Glyph="&#xEE3F;" />
                 </MenuFlyoutItem.Icon>
             </MenuFlyoutItem>
+            <MenuFlyoutItem
+                x:Name="RunAsAdminItem"
+                x:Uid="BaseLayoutItemContextRunAsAdminItem"
```
Comment (yaichenbaum) [inline Files/Views/LayoutModes/GridViewBrowser.xaml:235]:
```suggestion
                x:Name="RunAsAdmin"
                x:Uid="BaseLayoutItemContextRunAsAdmin"
```

File: Files/MultilingualResources/Files.de-DE.xlf [hunk 4]
```diff
@@ -861,6 +861,10 @@
           <source>B</source>
           <target state="new">B</target>
         </trans-unit>
+        <trans-unit id="BaseLayoutItemContextRunAsAdmin.Text" translate="yes" xml:space="preserve">
```
Comment (yaichenbaum) [inline Files/MultilingualResources/Files.de-DE.xlf:864]:
```suggestion
        <trans-unit id="BaseLayoutContextRunAsAdmin.Text" translate="yes" xml:space="preserve">
```

File: Files/Strings/en-US/Resources.resw [hunk 5]
```diff
@@ -753,4 +753,7 @@
   <data name="ByteSymbol" xml:space="preserve">
     <value>B</value>
   </data>
-</root>
\ No newline at end of file
+  <data name="BaseLayoutItemContextRunAsAdmin.Text" xml:space="preserve">
```
Comment (yaichenbaum) [inline Files/Strings/en-US/Resources.resw:756]:
```suggestion
  <data name="BaseLayoutContextFlyoutRunAsAdmin.Text" xml:space="preserve">
```

File: Files/Views/LayoutModes/GenericFileBrowser.xaml [hunk 6]
```diff
@@ -234,6 +234,16 @@
                     <FontIcon Glyph="&#xEE3F;" />
                 </MenuFlyoutItem.Icon>
             </MenuFlyoutItem>
+            <MenuFlyoutItem
+                x:Name="RunAsAdmin"
+                x:Uid="BaseLayoutItemContextRunAsAdmin"
```
Comment (yaichenbaum) [inline Files/Views/LayoutModes/GenericFileBrowser.xaml:239]:
```suggestion
                x:Uid="BaseLayoutContextFlyoutRunAsAdmin"
```

File: Files/Views/LayoutModes/GridViewBrowser.xaml [hunk 7]
```diff
@@ -230,6 +230,16 @@
                     <FontIcon Glyph="&#xEE3F;" />
                 </MenuFlyoutItem.Icon>
             </MenuFlyoutItem>
+            <MenuFlyoutItem
+                x:Name="RunAsAdmin"
+                x:Uid="BaseLayoutItemContextRunAsAdmin"
```
Comment (yaichenbaum) [inline Files/Views/LayoutModes/GridViewBrowser.xaml:235]:
```suggestion
                x:Uid="BaseLayoutContextFlyoutRunAsAdmin"
```

File: Files.Launcher/Program.cs [hunk 8]
```diff
@@ -335,7 +335,8 @@ private static void HandleApplicationLaunch(string application, AppServiceReques
                 process.StartInfo.FileName = application;
                 // Show window if workingDirectory (opening terminal)
                 process.StartInfo.CreateNoWindow = string.IsNullOrEmpty(workingDirectory);
-                process.StartInfo.Arguments = arguments;
+                if (arguments == "runas") process.StartInfo.Verb = "runas";
+                else process.StartInfo.Arguments = arguments;
```
Comment (tsvietOK) [inline Files.Launcher/Program.cs:339]:
```csharp
            if (arguments == "runas")
            {
                process.StartInfo.Verb = "runas";
            }
            else
            {
                process.StartInfo.Arguments = arguments;
            }
```

File: Files/Interacts/Interaction.cs [hunk 9]
```diff
@@ -173,19 +173,21 @@ public void GetPath_Click(object sender, RoutedEventArgs e)
             }
         }
 
-        public static async Task InvokeWin32Component(string applicationPath, string arguments = null)
+        public static async Task InvokeWin32Component(string applicationPath, string arguments = null, bool runAsAdmin = false)
```
Comment (tsvietOK) [inline Files/Interacts/Interaction.cs:176]: is `bool runAsAdmin` needed here?

File: Files/Interacts/Interaction.cs [hunk 10]
```diff
@@ -173,19 +173,21 @@ public void GetPath_Click(object sender, RoutedEventArgs e)
...
 
-        public static async Task InvokeWin32Components(List<string> applicationPaths, string arguments = null)
+        public static async Task InvokeWin32Components(List<string> applicationPaths, string arguments = null, bool runAsAdmin = true)
         {
             Debug.WriteLine("Launching EXE in FullTrustProcess");
             if (App.Connection != null)
             {
                 var value = new ValueSet();
+                value.Add("Application", ApplicationPath);
+                if (runAsAdmin) value.Add("Arguments", "runas");
```
Comment (tsvietOK) [inline Files/Interacts/Interaction.cs:188]:
```csharp
            if (runAsAdmin)
            {
                value.Add("Arguments", "runas");
            }
            else
            {
                value.Add("Arguments", arguments);
            }
```

**Comment by Jaiganeshkumaran** [comment]
Please don't merge yet.

**Comment by Jaiganeshkumaran** [comment]
You can merge now. I've also added the option to run as another user. This option is hidden in File Explorer however I included it by default in Files UWP since it'd be helpful for users who have multiple user accounts.

**Comment by yaichenbaum** [comment]
:shipit:

File: Files.Launcher/Program.cs [hunk 11]
```diff
@@ -343,7 +343,30 @@ private static void HandleApplicationLaunch(string application, AppServiceReques
                 process.StartInfo.FileName = application;
                 // Show window if workingDirectory (opening terminal)
                 process.StartInfo.CreateNoWindow = string.IsNullOrEmpty(workingDirectory);
-                process.StartInfo.Arguments = arguments;
+                if (arguments == "runas")
+                {
+                    process.StartInfo.UseShellExecute = true;
+                    process.StartInfo.Verb = "runas";
+                    if (Path.GetExtension(application) == ".msi")
```
Comment (duke7553) [inline Files.Launcher/Program.cs:350]: Shouldn't we check this in a case sensitive way?

**Pull request merged**